optimum/habana/transformers/trainer.py

```diff
@@ -1637,7 +1562,9 @@ def save_model(self, output_dir: Optional[str] = None, _internal_call: bool = Fa
             output_dir = self.args.output_dir
 
         if self.is_fsdp_enabled:
-            if "FULL_STATE_DICT" in str(self.accelerator.state.fsdp_plugin.state_dict_type):
+            if ("FULL_STATE_DICT" in str(self.accelerator.state.fsdp_plugin.state_dict_type)) and (
+                is_accelerate_available("0.24.2")
+            ):
```

Comment:
Is there a better way to do this than hardcode the version here? It will no doubt need to be changed in the future.

Reply:
I agree, but it is following the upstream code in [transformers](https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/src/transformers/trainer.py#L3600) and my goal is to make the upgrade easier for the next version 4.46.1 which has lots of changes

Reply:
I think it's safe to remove the version check as a more recent Accelerate version is required anyway: https://github.com/huggingface/optimum-habana/blob/ba914504bd52b989d05a4e11916aa19b3ce176f4/setup.py#L35